kratos/python_scripts/analysis_stage.py

```diff
@@ -344,6 +349,30 @@ def __CreateListOfProcesses(self):
         self._list_of_output_processes = self._CreateProcesses("output_processes", order_processes_initialization)
         self._list_of_processes.extend(self._list_of_output_processes) # Adding the output processes to the regular processes
 
+    def _GetMapOfAdditionalHistoricalVariables(self):
+        """This function returns the list of (additional) historicsl variables
+        needed in the simulation
+        It can be overridden in derived classes
+        """
+        map_additional_vars = defaultdict(list)
+        self.__GetLMapOfAdditionalVariablesFromProcesses("processes", map_additional_vars)
+        self.__GetLMapOfAdditionalVariablesFromProcesses("output_processes", map_additional_vars)
+
+        return map_additional_vars
+
+    def __GetLMapOfAdditionalVariablesFromProcesses(self, parameter_name, map_additional_vars):

```

Comment:
Since you are allowing `_GetMapOfAdditionalHistoricalVariables` to be overriden, may be we can remove name mangling here so derrived classes also can use the same method if required?

Reply:
goof point